lib/client-assets.php

```diff
@@ -699,6 +699,10 @@ function gutenberg_extend_block_editor_settings_with_default_editor_styles( $set
  */
 function gutenberg_extend_block_editor_settings_with_fse_theme_flag( $settings ) {
 	$settings['isFSETheme'] = gutenberg_is_fse_theme();
+
+	// Enable the new layout options for themes with a theme.json file.
+	$settings['supportsLayout'] = WP_Theme_JSON_Resolver::theme_has_support();
```

Comment:
Unrelated, but it feels we should name this `WP_Theme_Config_Resolver` not `JSON`

Reply:
cc @nosolosw

Reply:
There's also the class [WP_Theme_JSON](https://github.com/WordPress/gutenberg/blob/trunk/lib/class-wp-theme-json.php#L12), should we also rename it to `WP_Theme_Config` for coherence?

To be clear, I'm not 100% sold on these classes having the "theme" infix either. It is too theme-specific. Both classes process data that comes from core ([see](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental-default-theme.json)) and the user (stored in the database as CPT) as well. I guess I've come not to like that we call "theme.json" to a format that is shared across three different origins: core, theme, and user.

Happy to rename to anything, but thought I could try my luck and see if anyone sees a better name for both classes :)